src/main/java/com/mcmoddev/lib/data/Names.java

```diff
@@ -1,77 +1,46 @@
-        public static final String SMALLPOWDER = "smallpowder";
-        public static final String SWORD       = "sword";
-        public static final String GEM         = "gem";
+	ANVIL,ARROW,AXE,BLEND,BOOTS,BOLT,BOW,CHESTPLATE,CRACKHAMMER,CROSSBOW,DOOR,FISHINGROD,GEAR,HELMET,HOE,HORSEARMOR,INGOT,LEGGINGS,NUGGET,PICKAXE,POWDER,ROD,SHEARS,SHIELD,SHOVEL,SLAB,SMALLBLEND,SMALLPOWDER,SWORD,GEM,
 
         // Blocks
-        public static final String ANVILBLOCK    = "anvilBlock";
-        public static final String BARS          = "bars";
-        public static final String BLOCK         = "block";
-        public static final String BOOKSHELF     = "bookshelf";
-        public static final String BUTTON        = "button";
-        public static final String DOORBLOCK     = "doorBlock";
-        public static final String DOUBLESLAB    = "doubleSlab";
-        public static final String HALFSLAB      = "halfSlab";
-        public static final String FLOWERPOT     = "flowerPot";
-        public static final String LADDER        = "ladder";
-        public static final String LEVER         = "lever";
-        public static final String PLATE         = "plate";
-        public static final String PRESSUREPLATE = "pressurePlate";
-        public static final String STAIRS        = "stairs";
-        public static final String TRAPDOOR      = "trapdoor";
-        public static final String TRIPWIREHOOK  = "tripwireHook";
-        public static final String WALL          = "wall";
-        public static final String FENCE         = "fence";
-        public static final String FENCEGATE     = "fenceGate";
-
-        // Fluid stuff
-        public static final String FLUID      = "fluid";
-        public static final String FLUIDBLOCK = "fluidBlock";
+	ANVILBLOCK,BARS,BLOCK,BOOKSHELF,BUTTON,DOORBLOCK,DOUBLESLAB,HALFSLAB,FLOWERPOT,LADDER,LEVER,PLATE,PRESSUREPLATE,STAIRS,TRAPDOOR,TRIPWIREHOOK,WALL,FENCE,FENCEGATE,
 
         // Mekanism
-        public static final String CRYSTAL     = "crystal";
-        public static final String SHARD       = "shard";
-        public static final String CLUMP       = "clump";
-        public static final String POWDERDIRTY = "powderDirty";
+	CRYSTAL,SHARD,CLUMP,POWDERDIRTY,
 
         // IC2
-        public static final String CASING          = "casing";
-        public static final String DENSEPLATE      = "densePlate";
-        public static final String CRUSHED         = "crushed";
-        public static final String CRUSHEDPURIFIED = "crushedPurified";
+	CASING,DENSEPLATE,CRUSHED,CRUSHEDPURIFIED,
 
         // Ores
-        public static final String ORE       = "ore";
-        public static final String ENDORE    = "oreEnd";
-        public static final String NETHERORE = "oreNether";
+	ORE,ENDORE,NETHERORE;
+
+	
+    public static final TreeMap<Integer, Names> MAP = Maps.newTreeMap();
+
+    static {
+        for(Names name : values())
+            MAP.put(name.ordinal(),name);

```

Comment:
On  MAP.put(stat.ordinal(),stat);

not
        for(MaterialStats stat : values())